VideoLocker/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -1,32 +1,51 @@
 package org.edx.mobile.base;
 
 
+import android.app.Activity;
 import android.app.Application;
+import android.content.Intent;
 import android.content.IntentFilter;
 import android.graphics.Bitmap.CompressFormat;
+import android.os.Bundle;
 
 import com.crashlytics.android.Crashlytics;
 import com.newrelic.agent.android.NewRelic;
 
 import org.edx.mobile.logger.Logger;
 import org.edx.mobile.module.analytics.SegmentFactory;
+import org.edx.mobile.module.prefs.PrefManager;
 import org.edx.mobile.module.storage.Storage;
 import org.edx.mobile.receivers.NetworkConnectivityReceiver;
+import org.edx.mobile.util.AppConstants;
 import org.edx.mobile.util.Config;
 import org.edx.mobile.util.Environment;
 import org.edx.mobile.util.NetworkUtil;
 import org.edx.mobile.util.images.ImageCacheManager;
 import org.edx.mobile.util.images.RequestManager;
+import org.edx.mobile.view.Router;
+
+import java.util.ArrayList;
+import java.util.List;
 
 import io.fabric.sdk.android.Fabric;
 
 /**
  * This class initializes the modules of the app based on the configuration.
  */
-public class MainApplication extends Application {
+public class MainApplication extends Application{
+
+    protected final Logger logger = new Logger(getClass().getName());
 
     NetworkConnectivityReceiver connectivityReceiver;
 
+    private static MainApplication application;
+
+    public static final MainApplication instance(){
```

Comment:
@aleffert  :  I will rename it.   The rationale behind this public static method is that - we have several singletons across app, but Application class is a "singleton" and works as a factory,  it is easier for other singletons to be referenced through Application instance